host-bin/edit-chroot

```diff
@@ -385,8 +434,33 @@ for NAME in "$@"; do
         fi
         echo -n "  Restoring $src to $CHROOT..." 1>&2
         mkdir -p "$CHROOT"
-        spinner 1 tar --checkpoint=200 --checkpoint-action=exec=echo \
-            --one-file-system -xaf "$src" -C "$CHROOT" --strip-components=1
+        if [ -f "$src.part01" ]; then
+            # Detect the type of compression before sending it through a fifo
+            for tarparam in -z -j -J -Z --no-auto-compress fail; do
+                if [ "$tarparam" = 'fail' ]; then
+                    error 2 "Unable to detect compression method of $src"
+                elif tar $tarparam --test-label -f "$src" >/dev/null 2>&1; then
```

Comment:
If it fails here, it would fail in the one or two times we do the same check above (release detection, validity check, etc).
If we're concerned that someone will do an -s that results in smaller than one block, we should limit the minimum -s to something reasonable.

Actually, we should just have -s specify in megabytes.